cdk_python/src/detectors/python-cdk-auto-scaling-group-cooldown-period/python-cdk-auto-scaling-group-cooldown-period_non-compliant.py

```diff
@@ -0,0 +1,21 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+# {fact rule=python-cdk-auto-scaling-group-cooldown-period@v1.0 defects=1}
+import aws_cdk as cdk
+from aws_cdk import Vpc, Duration
+from aws_cdk.aws_autoscaling import AutoScalingGroup
+from aws_cdk.aws_ec2 import InstanceClass, InstanceType, MachineImage, Vpc
+
+class CdkStarter(cdk.Stack):
+    def __init__(self, scope: cdk.App, id: str):
+        super(scope, id)
+
+        # Noncompliant: No cooldown, leading to potential rapid, uncoordinated scaling.
+        AutoScalingGroup(self, 'rAsg',
+            vpc=Vpc(self, 'rVpc'),
+            instance_type=InstanceType(InstanceClass.T3),
+            machine_image=MachineImage.latestAmazonLinux(),
+            cooldown=Duration.seconds(0)
```

Comment:
in the Noncompliant code example, do we need to setup up the cooldown as `0` seconds. it this a required filed of ASG, can we have the noncompliant code example to not setup the cooldown.

Reply:
If we don't specify a cooldown period in the non-compliant code, the default value of 5 minutes will be applied, which will make it compliant. However, we can set the cooldown period to be less than the default, which would cause the non-compliant case to become problematic or buggy.
Doc link: https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_autoscaling/AutoScalingGroup.html